### PR TITLE
Add KGE and KGE' scores for hydrological verification

### PR DESCRIPTION
**Pull request by andreas-grafberger**
### Description

This proposes the addition of two new scores `kge` and `kge_prime` to `earthkit.meteo.score`. I’ll be using `kge_prime` for another project and thought they’d be a nice addition to earthkit-meteo.

The two implementations are based on the respective papers in which they were introduced:
- KGE: https://doi.org/10.1016/j.jhydrol.2009.08.003
- KGE': https://doi.org/10.1016/j.jhydrol.2012.01.011

The formula for the original KGE can also be found [on wikipedia](https://en.wikipedia.org/wiki/Kling%E2%80%93Gupta_efficiency) and a reference implementation for KGE' can be found [here](https://github.com/ecmwf/hat/blob/6ad904246fa75c3dbf453459961fabf1c2dccb3a/hat/compute_hydrostats/stats.py#L61C1-L68C67). 

I included a few test cases for both that document behavior in edge cases (e.g., zero mean or zero variance in one of the input arrays) that might produce unexpected results.

A few notes on the implementation:
- Small shared base class to centralise input checks and NaN handling (raise/propagate/omit).
- Simulation and observation inputs must be 2D (n_points, n_samples). Passing 1D arrays (n_samples,) is not supported and users must e.g. use `x[None, :]` and `y[None, :]`.
- All KGE components can be returned by setting `return_components=True`.
- Array API friendly; tested locally with NumPy and PyTorch.

### Contributor Declaration

By opening this pull request, I affirm the following:

* All authors agree to the [Contributor License Agreement](https://github.com/ecmwf/codex/blob/main/Legal/contributor_license_agreement.md).
* The code follows the project's coding standards.
* I have performed self-review and added comments where needed.
* I have added or updated tests to verify that my changes are effective and functional.
* I have run all existing tests and confirmed they pass.

### CODE REVIEW

File: src/earthkit/meteo/score/kge.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#
+
+from . import array
+
+
+def kge(*args, **kwargs):
+    return array.kge(*args, **kwargs)
+
+
+def kge_prime(*args, **kwargs):
```
Comment (corentincarton) [inline src/earthkit/meteo/score/kge.py:17]: @andreas-grafberger modified_kge instead? Can you ask the hydro people for a good name? I feel that prime won't be clear enough.

Comment (andreas-grafberger) [reply src/earthkit/meteo/score/kge.py:17]: @corentincarton Yeah, this is a bit tricky. I asked around a bit, and out of the three responses from researchers in our hydro team, I got two votes for kge_prime and one vote saying that either kge_prime or kge_modified would work. But overall, it seemed like nobody had a very strong opinion, as the naming conventions in other tools vary quite a lot anyway.

I don't mind either way, both `kge_prime` or `kge_modified` would work for me. :)

File: src/earthkit/meteo/score/array/kge.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+        if x.ndim != 2 or y.ndim != 2:
+            # Support 1D inputs would be a nice improvement but needs intuitive
+            # behavior when used with nan_policy="omit".
+            raise ValueError(
+                "x and y must be 2D arrays with shape (n_points, n_samples). "
+                "For a single time series, use x[None, :] and y[None, :]."
+            )
+
+        isnan_mask = xp.any(xp.isnan(x) | xp.isnan(y), axis=1)
```
Comment (Oisin-M) [inline src/earthkit/meteo/score/array/kge.py:57]: More of a design question, but should we have `axis=1` here? This means if for any point we have a single missing temporal sample, then we omit the full timeseries for that point

**Comment by andreas-grafberger** [comment]
Temporarily closing this PR as it needs follow up work.

**Comment by Oisin-M** [comment]
Work on adding xarray implementations of metrics including KGE ongoing in #94
